_internals/tds-protocol.md

```diff
@@ -85,11 +85,31 @@ Once we are over the <code>PRELOGIN</code> phase, we are ready for
 <code>LOGIN7</code> which defines the authentication rules for use 
 between client and server.
 
+### SSL/TLS support for TDS connections
 
-### Defining authentication rules.
+Babelfish uses native PostgreSQL support for using SSL connections to encrypt client/server communications. See PostgreSQL [documentation](https://www.postgresql.org/docs/13/ssl-tcp.html#SSL-SETUP) for details about configuring server-side SSL functionality.
+
+To use SSL encryption, [compile the PostgreSQL server with support for SSL,](https://www.postgresql.org/docs/13/install-procedure) and set the <code>ssl</code> parameter to <code>on</code> in the postgresql.conf file. The server will listen for SSL and non-SSL connections on the same TCP port, and will negotiate with any connecting client about using SSL. By default, a client can choose whether to require an SSL connection or not. The server will listen for both unencrypted and encrypted connections on the same TCP port, and negotiate the connection type with the connecting client.  If <code>babelfishpg_tds.tds_ssl_encrypt</code> is set to <code>true</code>, end-to-end encryption is enforced for all connections. In that case, if a client requests an unencrypted connection, the connection is rejected.
+
+The following table shows how Babelfish interprets SSL settings when a client connects:
+
+| Client SSL setting   | Babelfish SSL setting | Connection allowed?                         | Value returned to client |
+|:---------------------|:----------------------|:--------------------------------------------|:-------------------------|
+| ENCRYPT_OFF          | tds_ssl_encrypt=false | Allowed, the login packet is encrypted      | ENCRYPT_OFF              |
+| ENCRYPT_OFF          | tds_ssl_encrypt=true  | Allowed, the entire connection is encrypted | ENCRYPT_REQ              |
+| ENCRYPT_ON           | tds_ssl_encrypt=false | Allowed, the entire connection is encrypted | ENCRYPT_ON               |
+| ENCRYPT_ON           | tds_ssl_encrypt=true  | Allowed, the entire connection is encrypted | ENCRYPT_ON               |
+| ENCRYPT_NOT_SUP      | tds_ssl_encrypt=false | Yes                                         | ENCRYPT_NOT_SUP          |
+| ENCRYPT_NOT_SUP      | tds_ssl_encrypt=true  | No, connection closed                       | ENCRYPT_REQ              |
+| ENCRYPT_REQ          | tds_ssl_encrypt=false | Allowed, the entire connection is encrypted | ENCRYPT_ON               |
+| ENCRYPT_REQ          | tds_ssl_encrypt=true  | Allowed, the entire connection is encrypted | ENCRYPT_ON               |
+| ENCRYPT_CLIENT_CERT  | tds_ssl_encrypt=false | No, connection closed                       | Unsupported              |
+| ENCRYPT_CLIENT_CERT  | tds_ssl_encrypt=true  | No, connection closed                       | Unsupported              |
+
+### Defining authentication rules
 
 <code>LOGIN7</code> is the TDS ways of defining authentication rules. 
```

Comment:
/the TDS way of defining/is how TDS defines/

However, I don't understand what we're trying to say here. What `LOGIN7` is isn't terribly clear. More importantly, why does anyone even care? AFAICT `LOGIN7` is some super low-level detail of the TDS protocol, which doesn't seem like something users should need to care about. What am I missing here? When would a user need to understand `LOGIN7`?

Reply:
Yeah, I agree with you. We can remove this part. Let's address this in a separate pull request.